base/cartesian.jl

```diff
@@ -257,6 +257,56 @@ macro nif(N, condition, operation...)
     ex
 end
 
+"""
+    nif(condition, expression, ::Val{N}) where {N}
+    nif(condition, expression, else_expression, ::Val{N}) where {N}
+
+Generate a sequence of `if ... elseif ... else ... end` statements.
+
+# Arguments
+- `condition`: A function that takes an integer between `1` and `N` and
```

Comment:
```suggestion
- `condition`: A function that takes an integer between `1` and `N-1` and
```
If I understand the code correctly.